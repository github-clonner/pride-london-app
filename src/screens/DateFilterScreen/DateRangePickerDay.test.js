// @flow
import React from "react";
import { shallow } from "enzyme";
import DateRangePickerDay from "./DateRangePickerDay";
import * as dateLib from "../../lib/date";

const render = props =>
  shallow(
    <DateRangePickerDay onPress={() => {}} onLongPress={() => {}} {...props} />
  );

const date = (year, month, day) => {
  const d = new Date(Date.UTC(year, month, day));
  return {
    year,
    month,
    day,
    timestamp: d.getTime(),
    dateString: d.toISOString()
  };
};

describe("dateRangePickerDay", () => {
  it("renders without date selection", () => {
    const output = render({ date: date(2019, 7, 12), marking: {} });

    expect(output).toMatchSnapshot();
  });

  it("renders with a single selected day", () => {
    const marking = { selected: true, startingDay: true, endingDay: true };
    const output = render({ date: date(2019, 7, 12), marking });

    expect(output).toMatchSnapshot();
  });

  it("renders a beginning of a range", () => {
    const marking = { selected: true, startingDay: true };
    const output = render({ date: date(2019, 7, 12), marking });

    expect(output).toMatchSnapshot();
  });

  it("renders a beginning of a range", () => {
    const marking = { selected: true, endingDay: true };
    const output = render({ date: date(2019, 7, 12), marking });

    expect(output).toMatchSnapshot();
  });

  it("renders a middle of a range", () => {
    const marking = { selected: true };
    const output = render({ date: date(2019, 7, 12), marking });

    expect(output).toMatchSnapshot();
  });

  it("renders a marked day", () => {
    const marking = { marked: true };
    const output = render({ date: date(2019, 7, 12), marking });

    expect(output).toMatchSnapshot();
  });

  it("renders a marked, selected day", () => {
    const marking = { marked: true, selected: true };
    const output = render({ date: date(2019, 7, 12), marking });

    expect(output).toMatchSnapshot();
  });

  it("renders a disabled day if before today", () => {
    jest
      .spyOn(dateLib, "now")
      .mockImplementation(() => "2019-08-13T17:42:06+01:00");

    const output = render({
      date: date(2019, 7, 12),
      marking: {}
    });

    expect(output.props().disabled).toEqual(true);
    expect(output.props().accessibilityTraits).toContain("disabled");
    expect(output).toMatchSnapshot();
  });

  it("does not render a disabled day if same day as today", () => {
    jest
      .spyOn(dateLib, "now")
      .mockImplementation(() => "2019-08-12T17:42:06+01:00");

    const output = render({
      date: date(2019, 7, 12),
      marking: {}
    });

    expect(output.props().disabled).toEqual(false);
    expect(output.props().accessibilityTraits).not.toContain("disabled");
    expect(output).toMatchSnapshot();
  });

  describe("interaction", () => {
    it("responds to press", () => {
      const onPress = jest.fn();
      const output = render({ date: date(2019, 7, 12), onPress, marking: {} });

      output.simulate("press");

      expect(onPress).toHaveBeenCalled();
    });

    it("responds to long press", () => {
      const onLongPress = jest.fn();
      const output = render({
        date: date(2019, 7, 12),
        onLongPress,
        marking: {}
      });

      output.simulate("longPress");

      expect(onLongPress).toHaveBeenCalled();
    });
  });

  describe("#shouldComponentUpdate", () => {
    const props = {
      date: date(2019, 7, 12),
      marking: { selected: true, startingDay: true, endingDay: true }
    };

    it("stops update if state markings and date are the same", () => {
      const nextProps = {
        date: date(2019, 7, 12),
        marking: { selected: true, startingDay: true, endingDay: true },
        disabled: false
      };

      const output = render(props);
      const shouldUpdate = output.instance().shouldComponentUpdate(nextProps);

      expect(shouldUpdate).toBe(false);
    });

    it("updates when markings change", () => {
      const nextProps = {
        date: date(2019, 7, 12),
        marking: { selected: true, endingDay: true }
      };

      const output = render(props);
      const shouldUpdate = output.instance().shouldComponentUpdate(nextProps);

      expect(shouldUpdate).toBe(true);
    });

    it("updates when date changes", () => {
      const nextProps = {
        date: date(2019, 8, 12),
        marking: { selected: true, startingDay: true, endingDay: true }
      };

      const output = render(props);
      const shouldUpdate = output.instance().shouldComponentUpdate(nextProps);

      expect(shouldUpdate).toBe(true);
    });
  });
});
