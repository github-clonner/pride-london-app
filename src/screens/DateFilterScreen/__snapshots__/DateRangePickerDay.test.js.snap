// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dateRangePickerDay does not render a disabled day if same day as today 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityLabel="Monday, August 12"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  testID="calendar-day-2019-7-12"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(44, 218, 157)",
            "borderRadius": 17,
            "height": 35,
            "width": 34,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    >
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Array [
            Object {
              "marginTop": 7,
              "textAlign": "center",
            },
            Object {},
            Object {},
          ]
        }
        type="small"
      >
        12
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`dateRangePickerDay renders a beginning of a range 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityLabel="Monday, August 12"
  accessibilityTraits={
    Array [
      "button",
      "selected",
    ]
  }
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  testID="calendar-day-2019-7-12"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "rgb(44, 218, 157)",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(44, 218, 157)",
            "borderRadius": 17,
            "height": 35,
            "width": 34,
          },
          Object {
            "backgroundColor": "rgb(44, 218, 157)",
          },
        ]
      }
    >
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Array [
            Object {
              "marginTop": 7,
              "textAlign": "center",
            },
            Object {
              "color": "rgb(45, 47, 127)",
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
        type="small"
      >
        12
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`dateRangePickerDay renders a beginning of a range 2`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityLabel="Monday, August 12"
  accessibilityTraits={
    Array [
      "button",
      "selected",
    ]
  }
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  testID="calendar-day-2019-7-12"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "rgb(44, 218, 157)",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(44, 218, 157)",
            "borderRadius": 17,
            "height": 35,
            "width": 34,
          },
          Object {
            "backgroundColor": "rgb(44, 218, 157)",
          },
        ]
      }
    >
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Array [
            Object {
              "marginTop": 7,
              "textAlign": "center",
            },
            Object {
              "color": "rgb(45, 47, 127)",
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
        type="small"
      >
        12
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`dateRangePickerDay renders a disabled day if before today 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityLabel="Monday, August 12"
  accessibilityTraits={
    Array [
      "button",
      "disabled",
    ]
  }
  disabled={true}
  onLongPress={[Function]}
  onPress={[Function]}
  testID="calendar-day-2019-7-12"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(44, 218, 157)",
            "borderRadius": 17,
            "height": 35,
            "width": 34,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    >
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Array [
            Object {
              "marginTop": 7,
              "textAlign": "center",
            },
            Object {},
            Object {
              "color": "#cecece",
            },
          ]
        }
        type="small"
      >
        12
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`dateRangePickerDay renders a marked day 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityLabel="Monday, August 12"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  testID="calendar-day-2019-7-12"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(44, 218, 157)",
            "borderRadius": 17,
            "height": 35,
            "width": 34,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    >
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Array [
            Object {
              "marginTop": 7,
              "textAlign": "center",
            },
            Object {},
            Object {},
          ]
        }
        type="small"
      >
        12
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 2,
              "height": 4,
              "marginBottom": 5,
              "width": 4,
            },
            Object {
              "backgroundColor": "rgb(44, 218, 157)",
            },
          ]
        }
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`dateRangePickerDay renders a marked, selected day 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityLabel="Monday, August 12"
  accessibilityTraits={
    Array [
      "button",
      "selected",
    ]
  }
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  testID="calendar-day-2019-7-12"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "rgb(44, 218, 157)",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "rgb(44, 218, 157)",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(44, 218, 157)",
            "borderRadius": 17,
            "height": 35,
            "width": 34,
          },
          Object {
            "backgroundColor": "rgb(44, 218, 157)",
          },
        ]
      }
    >
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Array [
            Object {
              "marginTop": 7,
              "textAlign": "center",
            },
            Object {
              "color": "rgb(45, 47, 127)",
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
        type="small"
      >
        12
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 2,
              "height": 4,
              "marginBottom": 5,
              "width": 4,
            },
            Object {
              "backgroundColor": "rgb(45, 47, 127)",
            },
          ]
        }
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`dateRangePickerDay renders a middle of a range 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityLabel="Monday, August 12"
  accessibilityTraits={
    Array [
      "button",
      "selected",
    ]
  }
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  testID="calendar-day-2019-7-12"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "rgb(44, 218, 157)",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "rgb(44, 218, 157)",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(44, 218, 157)",
            "borderRadius": 17,
            "height": 35,
            "width": 34,
          },
          Object {
            "backgroundColor": "rgb(44, 218, 157)",
          },
        ]
      }
    >
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Array [
            Object {
              "marginTop": 7,
              "textAlign": "center",
            },
            Object {
              "color": "rgb(45, 47, 127)",
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
        type="small"
      >
        12
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`dateRangePickerDay renders with a single selected day 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityLabel="Monday, August 12"
  accessibilityTraits={
    Array [
      "button",
      "selected",
    ]
  }
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  testID="calendar-day-2019-7-12"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": 35,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(44, 218, 157)",
            "borderRadius": 17,
            "height": 35,
            "width": 34,
          },
          Object {
            "backgroundColor": "rgb(44, 218, 157)",
          },
        ]
      }
    >
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Array [
            Object {
              "marginTop": 7,
              "textAlign": "center",
            },
            Object {
              "color": "rgb(45, 47, 127)",
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
        type="small"
      >
        12
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`dateRangePickerDay renders without date selection 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityLabel="Monday, August 12"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  testID="calendar-day-2019-7-12"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(44, 218, 157)",
            "borderRadius": 17,
            "height": 35,
            "width": 34,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    >
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Array [
            Object {
              "marginTop": 7,
              "textAlign": "center",
            },
            Object {},
            Object {},
          ]
        }
        type="small"
      >
        12
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;
